src/Microsoft.Tye.Hosting.Diagnostics/LoggingSink.cs

```diff
@@ -26,7 +26,7 @@ public LoggingSink(Microsoft.Extensions.Logging.ILogger logger, DiagnosticsProvi
 
         public IDisposable Attach(EventPipeEventSource source, ReplicaInfo replicaInfo)
         {
-            using var loggerFactory = LoggerFactory.Create(builder => ConfigureLogging(replicaInfo.Service, replicaInfo.Replica, builder));
+            var loggerFactory = LoggerFactory.Create(builder => ConfigureLogging(replicaInfo.Service, replicaInfo.Replica, builder));
```

Comment:
This was a bug. We'd dispose the logger factory when this method exits, but we want to dispose it when the event pipe is closed.